### PR TITLE
YD3xn15mx

### PR DESCRIPTION
**Pull request by Thunderbird2086**
## Description

A macro keyboard with 15 keys and 3 encoders  from AliExpress
The original firmware doesn't support RGB Matrix. 
The default key map is for Zoom specific. 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by sigprof** [comment]
There is already a PR for this board, but under a different name — `winry/winry315` (@peepeetee was able to find the actual board designer):

* #15894

As for your difficulties with encoders,  your firmware does not seem to enable the support for encoders properly (it just adds the encoder A/B pins into `DIRECT_PINS`, but encoders don't work like that). But from my experience the encoders on that board are really lousy — the large center one completely broke during the first hour of use (I needed to replace it myself, the replacement seems to work fine for now), and two smaller ones sometimes skip steps (replacing them might be harder, because they are SMD, and their footprint does not seem to be compatible with anything except some noname encoders available on AliExpress).

**Comment by Thunderbird2086** [comment]
> 

Thank you, @sigprof!   I will do more research, and  make it work properly! 👍

**Comment by Thunderbird2086** [comment]
@sigprof, probably I can send a PR to your repo with my key mappings 😉

File: keyboards/YD3xn15mx/YD3xn15mx.c [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+void suspend_power_down_kb(void) {
+    rgb_matrix_set_suspend_state(true);
+    suspend_power_down_user();
+}
+
+void suspend_wakeup_init_kb(void) {
+    rgb_matrix_set_suspend_state(false);
+    suspend_wakeup_init_user();
+}
```
Comment (drashna) [inline keyboards/YD3xn15mx/YD3xn15mx.c:87]: This is no longer needed, and is handled in core code now.
```suggestion
```

File: keyboards/YD3xn15mx/YD3xn15mx.h [hunk 2]
```diff
@@ -0,0 +1,31 @@
+// Copyright 2022 Thunderbird2086 (Thunderbird2086)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include "quantum.h"
+#include "via.h"
```
Comment (drashna) [inline keyboards/YD3xn15mx/YD3xn15mx.h:7]:
```suggestion
```

File: keyboards/YD3xn15mx/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,87 @@
+// Copyright 2022 Thunderbird2086 (Thunderbird2086)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include QMK_KEYBOARD_H
+
+// Defines names for use in layer keycodes and the keymap
+enum layer_names {
+  _ZOOM_M = 0,
```
Comment (drashna) [inline keyboards/YD3xn15mx/keymaps/default/keymap.c:8]:
```suggestion
  _ZOOM_M,
```

File: keyboards/YD3xn15mx/readme.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
+# YD3xn15mx
+
+![YD3xn15mx](https://i.imgur.com/RdrcBlb.jpg)
```
Comment (drashna) [inline keyboards/YD3xn15mx/readme.md:3]:
```suggestion
![YD3xn15mx](https://i.imgur.com/RdrcBlbl.jpg)
```

File: keyboards/YD3xn15mx/rules.mk [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+RGB_MATRIX_DRIVER = WS2812
+
+RGBLIGHT_SUPPORTED = yes
+RGB_MATRIX_SUPPORTED = yes
+
+AUDIO_ENABLE = no           # Audio output
+
+# Encoders
+ENCODER_ENABLE = no 		# Enable support for Rotary Encoder
+EXTRA_ENCODERS_ENABLE = no  # Enables 3 encoders
```
Comment (drashna) [inline keyboards/YD3xn15mx/rules.mk:34]: This doesn't do anything
```suggestion
```

File: keyboards/YD3xn15mx/rules.mk [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+# Debug options
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+
+NKRO_ENABLE = no            # Enable N-Key Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+
+# RGB Options
+#RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+#RGBLIGHT_ANIMATIONS = yes   # LED animations
```
Comment (drashna) [inline keyboards/YD3xn15mx/rules.mk:23]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]
please rename the folder and files to all be lower case.

**Review by drashna — APPROVED** [review]

File: keyboards/yd3xn15mx/rules.mk [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+RGB_MATRIX_ENABLE = yes
+RGB_MATRIX_DRIVER = WS2812
+
+RGBLIGHT_SUPPORTED = yes
+RGB_MATRIX_SUPPORTED = yes
+
+AUDIO_ENABLE = no           # Audio output
+
+# Encoders
+ENCODER_ENABLE = no 		# Enable support for Rotary Encoder
```
Comment (fauxpark) [inline keyboards/yd3xn15mx/rules.mk:32]:
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
NKRO_ENABLE = no            # Enable N-Key Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
AUDIO_ENABLE = no           # Audio output
RGB_MATRIX_ENABLE = yes
RGB_MATRIX_DRIVER = WS2812
RGBLIGHT_SUPPORTED = yes
RGB_MATRIX_SUPPORTED = yes
```

File: keyboards/yd3xn15mx/readme.md [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+* Hardware Supported: Macro Keyboard 15 Keys + 3 Knobs
+* Hardware Availability: [AliExpress](https://www.aliexpress.com/wholesale?catId=0&initiative_id=SB_20220214015758&SearchText=Macro+Keyboard+Knob+15Key+%2B+3+Knob)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make YD3xn15mx:default
+
+Flashing example for this keyboard:
+
+    make YD3xn15mx:default:flash
```
Comment (fauxpark) [inline keyboards/yd3xn15mx/readme.md:17]:
```suggestion
    make yd3xn15mx:default

Flashing example for this keyboard:

    make yd3xn15mx:default:flash
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by zvecr** [comment]
With #15894 merged, any additional changes will need to target `winry/winry315`.

**Pull request closed without merging**